File: Javascript/Algorithms/461.js
/**
 * @param {number} x
 * @param {number} y
 * @return {number}
 */
var hammingDistance = function(x, y) {
    let diff = x ^ y;
    let count = 0;
    while (diff){
        count += diff & 1;
        diff >>= 1;
    }
    return count;
};

File: Rust/Algorithms/461.rs
impl Solution {
    pub fn hamming_distance(x: i32, y: i32) -> i32 {
        let mut diff = x ^ y;
        let mut count = 0;
        while (diff > 0) {
            count += diff & 1;
            diff >>= 1;
        }
        count
    }
}
